### PR TITLE
params: add /whoami endpoint

### PR DESCRIPTION
**Pull request by rogpeppe**
This mirrors the endpoint in other services.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/idmclient/100/

File: client_generated.go [hunk 1]
```diff
@@ -132,3 +133,11 @@ func (c *client) VerifyToken(ctx context.Context, p *params.VerifyTokenRequest)
 	err := c.Client.Call(ctx, p, &r)
 	return r, err
 }
+
+// WhoAmI returns authentication information on the client that is
+// making the WhoAmI call.
```
Comment (urosj) [inline client_generated.go]: skip last WhoAmI maybe?

File: params/params.go [hunk 2]
```diff
@@ -257,3 +257,14 @@ type SetUserExtraInfoItemRequest struct {
...
+
+// WhoAmIRequest holds parameters for requesting the current user name.
+type WhoAmIRequest struct {
+	httprequest.Route `httprequest:"GET /v1/whoami"`
+}
+
+// WhoAmIResponse holds information on the currently
+// authenticated user.
+type WhoAmIResponse struct {
+	User string `json:"user"`
```
Comment (mhilton) [inline params/params.go]: If it was going to mirror charmstore it would include groups

Comment (rogpeppe) [inline params/params.go]: I decided to mirror jem instead. I'm still not sure whether we'll always want to support group enumeration.

File: client_generated.go (hunk 1 above)
Comment (rogpeppe) [inline client_generated.go]: done

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/idmclient-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/idmclient/101/

**Pull request merged**